components/supervisor/pkg/ports/exposed-ports.go

```diff
@@ -108,27 +120,51 @@ func (g *GitpodExposedPorts) Observe(ctx context.Context) (<-chan []ExposedPort,
 		for {
 			select {
 			case u := <-updates:
-				if u == nil {
+				res := getExposedPorts(u)
+				if res == nil {
 					return
 				}
-
-				res := make([]ExposedPort, len(u.Status.ExposedPorts))
-				for i, p := range u.Status.ExposedPorts {
-					res[i] = ExposedPort{
-						LocalPort: uint32(p.Port),
-						Public:    p.Visibility == "public",
-						URL:       p.URL,
-					}
-				}
-
-				reschan <- res
+				g.exposedPorts <- res
 			case <-ctx.Done():
 				return
 			}
 		}
 	}()
 
-	return reschan, errchan
+	return g.exposedPorts, errchan
+}
+
+func getExposedPorts(wsInstance *gitpod.WorkspaceInstance) []ExposedPort {
+	if wsInstance == nil {
+		return nil
+	}
+
+	res := make([]ExposedPort, len(wsInstance.Status.ExposedPorts))
+	for i, p := range wsInstance.Status.ExposedPorts {
+		res[i] = ExposedPort{
+			LocalPort: uint32(p.Port),
+			Public:    p.Visibility == "public",
+			URL:       p.URL,
+		}
+	}
+	return res
+}
+
+func (g *GitpodExposedPorts) TriggerUpdate(ctx context.Context) error {
+	if g.exposedPorts == nil {
+		return nil
+	}
+
+	wsInfo, err := g.C.GetWorkspace(ctx, g.WorkspaceID)
+	if err != nil {
+		return err
+	}
+	res := getExposedPorts(wsInfo.LatestInstance)
```

Comment:
I think it is better if we do sync on level of `APIoverJSONRPC` in such way we don't need to worry about changing semantic of its clients.

i.e. add `APIoverJSONRPC.SyncInstance`, you can look how we appliy similar logic in the frontend, there are 2 important things:
- avoid multiple concurrent syncs, i.e. last one wins for isntance like here: https://github.com/gitpod-io/gitpod/blob/b4fa0fc9ba8398def9723f2461853e3634b40f03/components/gitpod-protocol/src/gitpod-service.ts#L529-L549
- resolve conflicts between instance updates and sync, if sync seen newer data then we shoudl ignore new instance update, it can be done base on phases like here: https://github.com/gitpod-io/gitpod/blob/b4fa0fc9ba8398def9723f2461853e3634b40f03/components/gitpod-protocol/src/gitpod-service.ts#L568-L571